tasks/main-clevis.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Ensure required packages are installed
+  package:
+    name: "{{ __nbde_client_packages }}"
+    state: present
+
+
+- name: Prepare keyfiles, perform clevis operations and dispose of keyfiles
+  block:
+    - name: Ensure we transfer keyfiles
+      copy:
```

Comment:
the `tempfile` module could be useful. This way you would not have to define a __nbde_client_managed_dir . https://docs.ansible.com/ansible/latest/modules/tempfile_module.html#tempfile-module

Reply:
Thanks, I will look into using it instead.

Reply:
Here is an example from tlog - https://github.com/linux-system-roles/tlog/blob/master/tests/tests_default_wrapper.yml - you create the `tempfile` in a `block`, then use the `always` clause to remove it.

Reply:
Thanks, I am using `tempfile` now.